### PR TITLE
Align ordering of core node properties with i3

### PR DESCRIPTION
**Pull request by jpalus**
Try to better mimic JSON node structure produced by i3 which might be
relied on by already existing tools. In particular having "type" right
after "id" is quite handy for streaming high-performance JSON parsers
such as simdjson (which are handy for maintaining responsiveness on
resource constrained systems).

refer https://github.com/i3/i3/blob/ab2a22a78b25ad12fed2c177a34c44950795cf33/src/ipc.c#L338

### CODE REVIEW

File: sway/ipc-json.c [hunk 1]
```diff
@@ -23,6 +23,20 @@
 static const int i3_output_id = INT32_MAX;
 static const int i3_scratch_id = INT32_MAX - 1;
 
+static const char *ipc_json_node_type_description(enum sway_node_type l) {
```
Comment (Xyene) [inline sway/ipc-json.c:26]: Nit: let's use something more descriptive than `l` here (`l` is used for `sway_container_layout` elsewhere in this file, which is bad as-is; making it represent a `sway_node_type` here makes things more confusing).

**Review by Xyene — CHANGES_REQUESTED** [review]
Thanks for working on this. I'm generally happy to merge this since it cleans up this code a bit, but note that we don't -- and can't -- make any guarantees about node ordering with respect to i3. It's also possible that future changes will make the ordering differ again (especially as new i3 properties are introduced).

File: sway/ipc-json.c (hunk 1 above)
Comment (jpalus) [reply sway/ipc-json.c]: Renamed to `node_type`.

**Comment by jpalus** [comment]
Thanks for review! As for lack of guarantees that's perfectly understandable. I'll take my chances though :).

**Review by Xyene — APPROVED** [review]
Thanks!

**Pull request merged**